test/postControllerTest.js

```diff
@@ -0,0 +1,148 @@
+  describe('GET /', () => {
+    it('should return correct "posts" in promise', () => (
+      request(app)
+        .get('/')
+        .then(() => {
+          postService.getAllPosts().then((posts) => {
+            assert.equal(posts, 'expected');
+          });
+        })
+    ));
+  });
+  describe('GET /posts', () => {
+    it('should call method "postService.getAllPosts" with no arguments', () => (
+      request(app)
+        .get('/posts')
+        .then(() => {
+          sinon.assert.calledWith(postService.getAllPosts);
+        })
+    ));
+  });
+  describe('GET /posts', () => {
+    it('should return correct "posts" in promise', () => (
+      request(app)
+        .get('/posts')
+        .then(() => {
+          postService.getAllPosts().then((posts) => {
+            assert.equal(posts, 'expected');
+          });
+        })
+    ));
+  });
+  describe('GET /posts/:post_id', () => {
+    it('should call method "postService.getOnePost" with correct arguments', () => (
+      request(app)
+        .get('/posts/post_id')
+        .then(() => {
+          sinon.assert.calledWith(postService.getOnePost, 'post_id');
+        })
+    ));
+  });
+  describe('GET /posts/:post_id', () => {
+    it('should return correct "post" in promise', () => (
+      request(app)
+        .get('/post/post_id')
+        .then(() => {
+          postService.getOnePost().then((post) => {
+            assert.equal(post, 'expected');
+          });
+        })
+    ));
+  });
+  describe('POST /post_new', () => {
+    it('should call method "postService.createNewPost" with correct arguments', () => (
+      request(app)
+        .post('/post_new')
+        .type('form')
+        .send({ title: 'title', author: 'author', body: 'body' })
+        .then(() => {
+          sinon.assert.calledWith(postService.createNewPost, 'title', 'author', 'body');
+        })
+    ));
+  });
+
+  // Nav Tests
+  describe('GET /', () => {
+    it('should respond with homepage', (done) => {
+      request(app)
+        .get('/')
+        .expect(200)
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(/My First Blog/, done);
+    });
+  });
+  describe('GET /posts', () => {
+    it('should respond with homepage', (done) => {
+      request(app)
+        .get('/posts')
+        .expect(200)
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(/My First Blog/, done);
+    });
+  });
+  describe('GET /posts/:post_id', () => {
+    it('should respond with post detail', (done) => {
+      request(app)
+        .get('/posts/post_id')
+        .expect(200)
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(/My First Blog/, done);
+    });
+  });
+  describe('GET /post_new', () => {
+    it('should respond with new post page', (done) => {
+      request(app)
+        .get('/post_new')
+        .expect(200)
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(/New Post/, done);
```

Comment:
This is not a nav test

Reply:
Why is this not a nav test when the three GET tests above are?

Reply:
Because this isn't testing anything in `nav.ejs`

Reply:
Ahh, I had thought nav tests meant testing navigation

Reply:
No, because `.expect(/My First Blog/, done);` is checking for the text "My First Blog", which is defined in nav.ejs.

This line is checking for "New Post", which is defined in "post_new.ejs"